Sources/Controllers/OARootViewController.m

```diff
@@ -130,6 +131,8 @@ - (void) viewDidLoad
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(productsRestored:) name:OAIAPProductsRestoredNotification object:nil];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(requestPurchase:) name:OAIAPRequestPurchaseProductNotification object:nil];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(reachabilityChanged:) name:kReachabilityChangedNotification object:nil];
+    
+    _mapViewController = [OARootViewController instance].mapPanel.mapViewController;
```

Comment:
Calling [OARootViewController instance] at OARootViewController...
I would suggest to use mapPanel.mapViewController instead of class field.

Reply:
Not sure how to fix this one.

Reply:
Just drop it and use mapPanel.mapViewController where you need